parser/src/json/schema.rs

```diff
@@ -610,7 +611,14 @@ fn compile_type(ctx: &Context, tp: &str, schema: &HashMap<&str, &Value>) -> Resu
             get("minItems"),
             get("maxItems"),
             get("prefixItems"),
-            get("items"),
+            match (get("items"), get("additionalItems")) {
+                (Some(items), None) => Some(items),
+                (None, Some(additional_items)) => Some(additional_items),
+                (Some(_), Some(_)) => {
+                    bail!("Cannot specify both 'items' and 'additionalItems'")
+                }
+                _ => None,
+            },
```

Comment:
We could alternatively only look for the one that's valid under the detected draft? That being said, I don't think a lot of folks actually annotate their schema with the draft, so that may be kind of useless... thoughts @mmoskal ?